wallet/wallet.go

```diff
@@ -693,20 +697,14 @@ func (w *Wallet) ChangePassphrase(old, new []byte) error {
 // a given account. It returns true if atleast one address in the account was
 // used and false if no address in the account was used.
 func (w *Wallet) AccountUsed(account uint32) (bool, error) {
-	addrs, err := w.Manager.AllAccountAddresses(account)
-	if err != nil {
-		return false, err
-	}
-	for _, addr := range addrs {
-		used, err := addr.Used()
-		if err != nil {
-			return false, err
-		}
-		if used {
-			return true, nil
-		}
-	}
-	return false, nil
+	var used bool
+	var err error
+	_ = w.Manager.ForEachAccountAddress(account,
+		func(maddr waddrmgr.ManagedAddress) error {
+			used, err = maddr.Used()
+			return nil
```

Comment:
This throws away the error and re-assigns the possibly true used value instead of breaking out of the loop and returning true to the caller.